cmd/apiserver/app/server/run_server.go

```diff
@@ -33,9 +33,8 @@ import (
 // RunServer runs an API server with configuration according to opts
 func RunServer(opts *ServiceCatalogServerOptions, stopCh <-chan struct{}) error {
 	if stopCh == nil {
-		/* the caller of RunServer should generate the stop channel
-		if there is a need to stop the API server */
-		stopCh = make(chan struct{})
+		/* whe RespectsStopCh: true and the stop channel is nil, error returned*/
+		return fmt.Errorf("%v", "the stop channel can not be nil")
```

Comment:
An error is good enough, but I still think this should be a panic. it's unreachable in normal case. If someone is going to library-ize this code and use it in their own apiserver, they need to know to set it.

Reply:
yes, i will change the code.

Reply:
@MHBauer i've changed the code comment as it is.